packages/nitro-protocol/contracts/examples/TwoOfThree.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.4;
+pragma experimental ABIEncoderV2;
+
+import '../interfaces/IForceMoveApp.sol';
+import '../Outcome.sol';
+import '../interfaces/IForceMove.sol';
+
+/**
+ * @dev The HashLockedSwap contract complies with the ForceMoveApp interface and implements a HashLockedSwaped payment
+ */
+contract TwoOfThree is IForceMoveApp2 {
+    struct AppData {
+        IForceMoveApp2 twoPartyApp;
+        bytes twoPartyAppData;
+    }
+
+    function validTransition(
+        VariablePart memory a, // TODO consider less confusing variable names
+        VariablePart memory b,
+        uint48 turnNumB,
+        uint256 nParticipants,
+        uint8 signedBy // Who has signed state b?
+    ) public override pure returns (bool) {
+        // If the new state is signed by all three participants, it is automatically considered a valid transition
+        if (signedBy == 7) return true;
+        if (signedBy == 0) return false;
+        if (signedBy < 4) {
```

Comment:
The joint (3 party channel) behaves a bit like the null app: unanimous signing implies a supported state (here, it implies a valid transition).

When participant 2 (the intermediary) has not signed, we fall through to another 2 party `ForceMoveApp.validTransition()`. That "inner" valid transition function can define any desired turn taking rules, by using `signedBy` and `turnNumB`.